File: sast-python2/case/completeness/single_app_tracing/expression/special_expression/template_literal_003_T.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = true
# evaluation item = 完整度->单应用跟踪完整度->表达式->语言特有的表达式
# scene introduction = 模板字面量
# level = 2+
# bind_url = completeness/single_app_tracing/expression/special_expression/template_literal_003_T
# evaluation information end
import os


def template_literal_003_T(taint_src):
    result = "{}_".format(taint_src)
    taint_sink(result)


def taint_sink(o):
    os.system(o)


if __name__ == '__main__':
    taint_src = "taint_src_value"
    template_literal_003_T(taint_src)

File: sast-js/case/completeness/single_app_tracing/expression/special_expression/template_literal_003_T.js
// evaluation information start
// real case = true
// evaluation item = 完整度->单应用跟踪完整度->表达式->语言特有的表达式
// scene introduction = 模板字面量
// level = 2
// bind_url = completeness/single_app_tracing/expression/special_expression/template_literal_003_T
// evaluation information end
const { execSync } = require('child_process');


function template_literal_003_T(__taint_src) {
  let result = `${__taint_src}_`;
  __taint_sink(result);
}

function __taint_sink(o) {
  execSync(o);
}

const taint_src = "taint_src_value";

template_literal_003_T(taint_src);
